### ISSUE TITLE
The digital human walks on tiptoe with the "Kimodo-SMPLX-RP-v1" model

### ISSUE THREAD

**Issue by sunmoon2018** [nv-tlabs/kimodo#18]
Hello, thank you for open source. I discovered a problem, when generating motion data using the "Kimodo-SMPLX-RP-v1" model, the digital human walks on tiptoe. See the following video for an example. However, this problem does not occur when using the "Kimodo-SOMA-RP-v1" or "Kimodo-SOMA-RP-v1" model. What could be the reason for this, and how can it be resolved?

Looking forward to your reply.

https://github.com/user-attachments/assets/fb33295c-6f86-4c6d-8e62-c79def084658

**Comment by davrempe** [comment]
This is likely due to the post-processing that is enabled by default in the demo. Post-processing uses IK to stick foot joints on the floor when the model predicts there is a foot contact, but if the predicted foot positions are high above the floor, it can result in the artifacts you're seeing. You can disable post-processing in the demo under Generate > Post Processing > Uncheck Enable. 

However, disabling post-processing will likely result in feet floating above the floor for this specific motion. Trying a different seed for generation (Generate > Model Parameters > Seed) may give better luck. But in general, the Kimodo-SMPLX model does tend to give lower-quality motions than Kimodo-SOMA, due to the retargeting process of the training data.

**Comment by sunmoon2018** [comment]
> This is likely due to the post-processing that is enabled by default in the demo. Post-processing uses IK to stick foot joints on the floor when the model predicts there is a foot contact, but if the predicted foot positions are high above the floor, it can result in the artifacts you're seeing. You can disable post-processing in the demo under Generate > Post Processing > Uncheck Enable.
> 
> However, disabling post-processing will likely result in feet floating above the floor for this specific motion. Trying a different seed for generation (Generate > Model Parameters > Seed) may give better luck. But in general, the Kimodo-SMPLX model does tend to give lower-quality motions than Kimodo-SOMA, due to the retargeting process of the training data.

@davrempe Thank you for your reply. That's true. I'm currently generating motion data using the Kimodo-SOMA model and saving it in NPZ or BVH format. However, I want to convert it to AMASS format via the "BVH → Kimodo NPZ → AMASS" route, but I'm getting an error when executing the "Kimodo NPZ → AMASS" step:
`(kimodo) wm@wm-ubantu:~/code_of_Ken/kimodo$ kimodo_convert   --from kimodo   --to amass   /home/wm/code_of_Ken/GMR/output_gvhmr_pt/kimodo_output_4.npz   /home/wm/code_of_Ken/GMR/output_gvhmr_pt/kimodo_output_44.npz
/home/wm/code_of_Ken/kimodo/kimodo/skeleton/base.py:72: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  neutral_joints = torch.load(pfolder / "joints.p").squeeze()
/home/wm/code_of_Ken/kimodo/kimodo/skeleton/base.py:76: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  bvh_neutral_joints = torch.load(pfolder / "bvh_joints.p").squeeze()
/home/wm/code_of_Ken/kimodo/kimodo/skeleton/base.py:81: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  global_rot_offsets = torch.load(global_offset_path).squeeze()
Error: Kimodo→AMASS requires 22 joints (SMPL-X); this file has J=77.`

So what should I do to get 22 joints Kimodo NPZ?
Additionally, my purpose in acquiring AMASS format data is to retarget it to my own robot using GMR.

**Comment by davrempe** [comment]
The AMASS format is specific to the Kimodo-SMPLX model, so you'll need to generate motion directly with Kimodo-SMPLX to retarget it with GMR (as shown in [the readme here](https://github.com/nv-tlabs/kimodo?tab=readme-ov-file#retargeting-motions-to-other-robots-with-gmr)). The SOMA skeleton is different from SMPL-X (different number of joints and different joint placements) and not supported by GMR.

**Comment by sunmoon2018** [comment]
> The AMASS format is specific to the Kimodo-SMPLX model, so you'll need to generate motion directly with Kimodo-SMPLX to retarget it with GMR (as shown in [the readme here](https://github.com/nv-tlabs/kimodo?tab=readme-ov-file#retargeting-motions-to-other-robots-with-gmr)). The SOMA skeleton is different from SMPL-X (different number of joints and different joint placements) and not supported by GMR.

Thank you for your reply. That means that although Kimodo-SOMA can obtain higher quality motion data than Kimodo-SMPLX, currently, if I want to use GMR, I can only use the Kimodo-SMPLX model, right? 

Additionally, in your document (as shown in the screenshot below), the kimodo_convert command is used to convert motion data formats. In the "Kimodo NPZ → AMASS" process, the "Kimodo NPZ" data can only come from the Kimodo-SMPLX model and cannot come from the Kimodo-SOMA model in the "BVH → Kimodo NPZ" process, is that correct?

<img width="800" height="352" alt="Image" src="https://github.com/user-attachments/assets/8903ee12-7f1c-4edc-be41-0bf91ad492a2" />

**Comment by davrempe** [comment]
> if I want to use GMR, I can only use the Kimodo-SMPLX model, right?

Yes, that's what I would recommend for now. You may also consider adapting the [SOMA retargeter](https://github.com/NVIDIA/soma-retargeter) to transfer motions from Kimodo-SOMA to your robot, or opening an issue there to request support for your robot.

> the "Kimodo NPZ" data can only come from the Kimodo-SMPLX model and cannot come from the Kimodo-SOMA

Yes exactly. [This table](https://research.nvidia.com/labs/sil/projects/kimodo/docs/user_guide/motion_convert.html#supported-conversions) in the docs outlines supported conversions and input/output requirements.

**Comment by sunmoon2018** [comment]
> > if I want to use GMR, I can only use the Kimodo-SMPLX model, right?
> 
> Yes, that's what I would recommend for now. You may also consider adapting the [SOMA retargeter](https://github.com/NVIDIA/soma-retargeter) to transfer motions from Kimodo-SOMA to your robot, or opening an issue there to request support for your robot.
> 
> > the "Kimodo NPZ" data can only come from the Kimodo-SMPLX model and cannot come from the Kimodo-SOMA
> 
> Yes exactly. [This table](https://research.nvidia.com/labs/sil/projects/kimodo/docs/user_guide/motion_convert.html#supported-conversions) in the docs outlines supported conversions and input/output requirements.

Okay, I understand, thank you very much. I will look into how to retarget the motion data generated by the Kimodo-SOMA model to my own robot.
